### DISCUSSION TITLE
I can't use files in Serverless Functions.

### DISCUSSION THREAD

**Discussion by makutamoto** [vercel/next.js#15842] [Help]
I'm creating a test application that uses files in Serverless Function.
Repository: [makutamoto/vercel_test](https://github.com/makutamoto/vercel_test)

According to [this](https://vercel.com/knowledge/how-can-i-use-files-in-serverless-functions), I created vercel.json like this:

```json
{
    "functions": {
        "pages/api/hello.tsx": {
            "includeFiles": "_files/*"
        }
    }
}
```

And created test application that tests whether files are uploaded or not.
```javascript
// pages/api/hello.tsx
// Next.js API route support: https://nextjs.org/docs/api-routes/introduction

import fs from 'fs';
import { NextApiRequest, NextApiResponse } from 'next'

const handler = (req: NextApiRequest, res: NextApiResponse) => {
  res.statusCode = 200;
  res.json(fs.readdirSync(__dirname));
}

export default handler
``` 

However, when I access api/hello, I get the following results.
```json
["bin","boot","dev","etc","home","lib","lib64","media","mnt","opt","proc","root","run","sbin","srv","sys","tmp","usr","var"]
```
It seams that files I specified so far are not uploaded to Serverless Function.

I want to know how to fix this problem.

Help.

**Comment by TooTallNate** [comment]
Note that the docs page that you reference [mentions](https://vercel.com/knowledge/how-can-i-use-files-in-serverless-functions#next.js) that `includeFiles` is not yet supported for Next.js API functions:

> ### Next.js
> At present, it is not possible to import files in a Next.js SSR page or function given [this issue](https://github.com/vercel/next.js/issues/8251).

**Reply by makutamoto** [reply]
Oh... I had missed that... Is there alternative way to achive file uploading?

**Comment by djedu28** [comment]
The error was also caused by using `__dirname`, instead use `process.cwd()`. The files listed in the error above are those in the root of the Linux system.

https://vercel.com/guides/how-can-i-use-files-in-serverless-functions
![image](https://github.com/user-attachments/assets/7d9cdbc6-3b8a-468c-8adf-057c53eead64)
